src/api/data-structure-creator.js

```diff
@@ -0,0 +1,100 @@
+export class DataStructureCreate {
+    constructor(name) {
+        if (!name) {
+            throw new Error('name is required.');
+        } else if (_isNotString(name)) {
+            throw new Error('name should be string');
+        } else {
+            this.name = name;
+        }
+        this.items = JSON.parse(localStorage.getItem(`${name}`) || "[]");
+    }
+    
+    createUser(data) {
+        let items = this.items;
+        let name = this.name;
+        if (!_isObject(data)) {
+            throw new Error('data should be an object.');
+        }
+        Object.defineProperty(data, '_id', {
+            enumerable: true,
+            configurable: false,
+            writable: false,
+            value: _generateId()
+        });
+        _validateUser(data);
+        items.push(data);
+        localStorage.setItem(name, JSON.stringify(items));
+        return data;
+    }
+
+    getUser(id) {
+        let items = this.items;
+        let user = {};
+        items.forEach((item) => {
+            if (item.id === id) {
+                return user = item;
+            }     
+        });
+        return user;
+    }
+
+    getUsers() {
+      return this.items;
+    }
+
+    updateUser(id, newUserData) {
+        let name = this.name;
+        let items = this.items;
+        let item = {};
+        if (!_isObject(newUserData)) {
+            console.log('data should be an object.');
+        }
+        if (Array.isArray(items) && items.length) {
+            items.forEach((item) => {
+                if (item.id === id) {
+                    Object.assign(item, newUserData);
+                    localStorage.setItem(name, JSON.stringify(items));
+                }
+            });
+        } else {
+            return item;
+        }
+    }
+}
+
+try {
+    localStorage.setItem('name', 'value');
+    localStorage.removeItem('name');
+} catch (e) {
+    throw new Error('LocalStorage is not supported');
+}
+
+function _generateId() {
+    return Math.random().toString(36).substr(2, 12);
+}
+
+function _validateUser(data) {
```

Comment:
Could clean this up a bit... just spit balling here:
```suggestion
function _validateUser(data) {
    const userProperties = [email, firstName, lastName, phone, _id];
    
    const incompleteProfile = userProperties.every(prop => Object.keys(data).contains(prop))
    const validValues = Object.values(data).every(value => typeof value === 'string');
    return !incompleteProfile && validValues;
}
```

I do know you more than likely won't want to rely on the length of an object... about ever. Hence me removing it from my suggestion.

Reply:
So much better!